src/app/features/reporting/query.service.ts

```diff
@@ -41,13 +41,19 @@ export class QueryService {
     query: string,
     from: Date = null,
     to: Date = null,
-    data: any = this.entities
+    data?: any
   ): Promise<any> {
-    if (!data || (data === this.entities && from < this.dataAvailableFrom)) {
+    if (from === null) {
```

Comment:
```suggestion
    if (!from) {
```
Just to be bullet proof